packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/CopyFromRealmTests.kt

```diff
@@ -0,0 +1,755 @@
+import io.realm.kotlin.query.RealmResults
+import io.realm.kotlin.schema.ListPropertyType
+import io.realm.kotlin.schema.RealmProperty
+import io.realm.kotlin.schema.RealmStorageType
+import io.realm.kotlin.schema.SetPropertyType
+import io.realm.kotlin.schema.ValuePropertyType
+import io.realm.kotlin.test.assertFailsWithMessage
+import io.realm.kotlin.test.platform.PlatformUtils
+import io.realm.kotlin.types.BaseRealmObject
+import io.realm.kotlin.types.MutableRealmInt
+import io.realm.kotlin.types.ObjectId
+import io.realm.kotlin.types.RealmInstant
+import io.realm.kotlin.types.RealmList
+import io.realm.kotlin.types.RealmSet
+import io.realm.kotlin.types.RealmUUID
+import org.mongodb.kbson.BsonObjectId
+import kotlin.reflect.KMutableProperty1
+import kotlin.reflect.KProperty1
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertContentEquals
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertNotEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertNotSame
+import kotlin.test.assertNull
+import kotlin.test.assertSame
+import kotlin.test.assertTrue
+import kotlin.test.fail
+
+class CopyFromRealmTests {
+
+    lateinit var tmpDir: String
+    lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(schema = setOf(Sample::class) + embeddedSchema)
+            .directory(tmpDir)
+            .build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun primitiveValues() { // This also checks that any default values set in the class are being overridden correctly.
+        val type = Sample::class
+        val schemaProperties = type.realmObjectCompanionOrThrow().io_realm_kotlin_schema().properties
+        val fields: Map<String, KProperty1<*, *>> = type.realmObjectCompanionOrThrow().io_realm_kotlin_fields
+
+        // Dynamically set data on the Sample object
+        val originalObject = Sample()
+        schemaProperties.forEach { prop: RealmProperty ->
+            if (prop.type is ValuePropertyType) {
+                val accessor: KMutableProperty1<BaseRealmObject, Any?> = fields[prop.name] as KMutableProperty1<BaseRealmObject, Any?>
+                val fieldValue: Any? = createPrimitiveValueData(accessor)
+                accessor.set(originalObject, fieldValue)
+            }
+        }
+
+        // Round-trip object through `copyToRealm` and `copyFromRealm`.
+        val unmanagedCopy = realm.writeBlocking {
+            copyToRealm(originalObject).copyFromRealm()
+        }
+
+        // Close Realm to ensure data is decoupled from Realm
+        realm.close()
```

Comment:
Maybe just assert that touching the `originalObject` is in fact throwing to highlight that we are not just linking back to the original 🤪